### PR TITLE
feat: add deno template

### PR DESCRIPTION
**Pull request by jacob-ebey**
This PR introduces a create-remix template option for Deno.

### CODE REVIEW

File: packages/create-remix/templates/deno/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Welcome to Remix!
+
+- [Remix Docs](https://remix.run/docs)
+
+## Development
+
+You will be utlizing Deno in watch mode. This is already wired up in your package.json as the `dev` script:
+
+```sh
+# start the remix dev server and wrangler
```
Comment (mjackson) [inline packages/create-remix/templates/deno/README.md:10]: Update comment

File: packages/create-remix/templates/deno/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "scripts": {
+    "postinstall": "remix setup deno",
+    "build": "cross-env NODE_ENV=production remix build",
+    "dev:remix": "cross-env NODE_ENV=development remix watch",
+    "dev:wrangler": "cross-env NODE_ENV=development deno run --watch --allow-net --allow-read --allow-env ./build/index.js",
```
Comment (mjackson) [inline packages/create-remix/templates/deno/package.json:6]: Update name of dev script here

File: packages/create-remix/templates/deno/server.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { serve } from "https://deno.land/std/http/server.ts";
+
+import { createRequestHandlerWithStaticFiles } from "@remix-run/deno";
+import * as build from "@remix-run/dev/server-build";
+
+const remixHandler = createRequestHandlerWithStaticFiles({
+  build,
+  mode: process.env.NODE_ENV
```
Comment (mjackson) [inline packages/create-remix/templates/deno/server.js:8]: Let's add a global variable that our compiler knows how to substitute during the build called `REMIX_MODE`.

File: packages/remix-deno/server.ts [hunk 4]
```diff
@@ -94,11 +94,7 @@ export function createRequestHandlerWithStaticFiles<Context = unknown>({
   return async (request: Request) => {
     try {
       return await serveStaticFiles(request, staticFiles);
-    } catch (e: any) {
-      if (e.code !== "EISDIR" && e.code !== "ENOENT") {
-        throw e;
-      }
-    }
+    } catch (e) {}
```
Comment (mjackson) [inline packages/remix-deno/server.ts:97]: Let's add a comment inside the `catch` that explains why we're not handling this.

File: packages/remix-deno/sessions/fileStorage.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        // (2^32). However, the larger id space should help to avoid collisions
+        // with existing ids when creating new sessions, which speeds things up.
+        let id = "";
+        for (let i = 0; i < randomBytes.length; ++i) {
+          id += ("0" + randomBytes[i].toString(16)).slice(-2);
+        }
+
+        try {
+          let file = getFile(dir, id);
+          // @ts-expect-error
```
Comment (mjackson) [inline packages/remix-deno/sessions/fileStorage.ts:51]: Let's add Deno's types so we can develop Remix and use them.

**Review by mjackson — CHANGES_REQUESTED** [review]

File: packages/create-remix/templates/deno/server.js (hunk 3 above)
Comment (mjackson) [reply packages/create-remix/templates/deno/server.js:8]: Actually, let's just put a TODO on this line and add this var in a separate PR.

**Comment by LANCENET** [comment]
Ok

File: packages/create-remix/templates/deno_ts/app/root.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width,initial-scale=1" />
+        <Meta />
+        <Links />
+      </head>
+      <body>
+        <Outlet />
+        <ScrollRestoration />
+        <Scripts />
+        {process.env.NODE_ENV === "development" && <LiveReload />}
```
Comment (MichaelDeBoey) [inline packages/create-remix/templates/deno_ts/app/root.tsx:29]:
```suggestion
        <LiveReload />
```
See #1352

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Pull request merged**